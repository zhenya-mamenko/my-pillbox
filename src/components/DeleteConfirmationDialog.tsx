import { type ComponentProps } from 'react';
import { StyleSheet, View } from 'react-native';
import Dialog from "react-native-dialog";
import t from '@/helpers/localization'
import useThemeColor from '@/hooks/useThemeColor';


interface Props extends ComponentProps<any> {
  showDialog: boolean;
  onClose: () => void;
  onDelete: () => void;
}

export default function DeleteConfirmationDialog({ showDialog, onClose, onDelete }: Props) {

  const styles = StyleSheet.create({
    deleteDialog: {
      alignItems: "center",
      flex: 1,
      justifyContent: "center",
    },
    deleteDialogContent: {
      backgroundColor: useThemeColor('background'),
      color: useThemeColor('text'),
    },
  });

  return (
    <View style={styles.deleteDialog}>
      <Dialog.Container
        headerStyle={styles.deleteDialogContent}
        contentStyle={styles.deleteDialogContent}
        footerStyle={styles.deleteDialogContent}
        visible={showDialog}
        onBackdropPress={onClose}
      >
        <Dialog.Title>{ t('delete_confirmation_header') }</Dialog.Title>
        <Dialog.Description>
          { t('delete_confirmation_description') }
        </Dialog.Description>
        <Dialog.Button
          color={useThemeColor('primary')}
          label={ t('cancel') }
          testID='cancel-button'
          onPress={onClose}
        />
        <Dialog.Button
          bold={true}
          color={useThemeColor('primary')}
          label={ t('delete') }
          testID='delete-button'
          onPress={onDelete}
        />
      </Dialog.Container>
    </View>
  );
}
