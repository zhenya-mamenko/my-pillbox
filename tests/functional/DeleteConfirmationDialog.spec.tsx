import { render, screen, userEvent, waitFor } from '@testing-library/react-native';
import DeleteConfirmationDialog from '@/components/DeleteConfirmationDialog';


global.requestAnimationFrame = jest.fn();

jest.mock('@/hooks/useThemeColor', () => (color: string) => {
  const colors: any = {
    background: '#ffffff',
    text: '#000000',
  }
  return colors[color];
});

jest.mock('@/helpers/localization', () => ({
  __esModule: true,
  default: (s: string) => s
}));

describe('DeleteConfirmationDialog', () => {

  it('Should render', () => {
    render(<DeleteConfirmationDialog showDialog={true} onClose={() => {}} onDelete={() => {}} />);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Should contain right data', async() => {
    render(<DeleteConfirmationDialog showDialog={true} onClose={() => {}} onDelete={() => {}} />);

    await waitFor(async () => {
      expect(await screen.getByText('delete_confirmation_description')).toBeTruthy();
      expect(await screen.getByText('delete_confirmation_header')).toBeTruthy();
    });
  });

  it('Should call onClose', async () => {
    const onClose = jest.fn();
    render(<DeleteConfirmationDialog showDialog={true} onClose={onClose} onDelete={() => {}} />);

    await waitFor(async () => {
      expect(await screen.getByText('delete_confirmation_header')).toBeTruthy();
    });

    const user = userEvent.setup();
    await user.press(screen.getByTestId('cancel-button'));

    expect(onClose).toHaveBeenCalled();
  });

  it('Should call onDelete', async () => {
    const onDelete = jest.fn();
    render(<DeleteConfirmationDialog showDialog={true} onClose={() => {}} onDelete={onDelete} />);

    await waitFor(async () => {
      expect(await screen.getByText('delete_confirmation_header')).toBeTruthy();
    });

    const user = userEvent.setup();
    await user.press(screen.getByTestId('delete-button'));

    expect(onDelete).toHaveBeenCalled();
  });

});
