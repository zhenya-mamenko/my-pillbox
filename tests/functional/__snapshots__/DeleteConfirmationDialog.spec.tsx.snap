// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirmationDialog Should render 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onHide={[Function]}
    renderToHardwareTextureAndroid={true}
    transparent={true}
    useNativeDriver={false}
    visible={true}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "black",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      collapsable={false}
      needsOffscreenAlphaCompositing={true}
      pointerEvents="box-none"
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0,
          "transform": [
            {
              "scale": 1.2,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          [
            {
              "marginTop": 22,
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 13,
                "flexDirection": "column",
                "overflow": "hidden",
                "width": 270,
              },
              {
                "backgroundColor": "#ffffff",
                "color": "#000000",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": {
                    "semantic": [
                      "systemGray6",
                    ],
                  },
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              [
                {
                  "margin": 18,
                },
                {
                  "backgroundColor": "#ffffff",
                  "color": "#000000",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": {
                      "semantic": [
                        "label",
                      ],
                    },
                    "fontSize": 18,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              delete_confirmation_header
            </Text>
            <Text
              style={
                [
                  {
                    "color": {
                      "semantic": [
                        "label",
                      ],
                    },
                    "fontSize": 13,
                    "marginTop": 4,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              delete_confirmation_description
            </Text>
          </View>
          <View
            style={
              {
                "backgroundColor": {
                  "semantic": [
                    "separator",
                  ],
                },
                "height": 0.5,
                "width": "100%",
              }
            }
          />
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                null,
                {
                  "backgroundColor": "#ffffff",
                  "color": "#000000",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 46,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "color": {
                        "semantic": [
                          "link",
                        ],
                      },
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    {
                      "color": "#007ff9",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
                testID="cancel-button"
              >
                cancel
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": {
                      "semantic": [
                        "separator",
                      ],
                    },
                    "height": "100%",
                    "width": 0.5,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 46,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "color": {
                        "semantic": [
                          "link",
                        ],
                      },
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    {
                      "color": "#007ff9",
                      "fontWeight": "600",
                    },
                    undefined,
                  ]
                }
                testID="delete-button"
              >
                delete
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
